src/features/game/events/landExpansion/triggerTornado.ts

```diff
@@ -0,0 +1,101 @@
+import { produce } from "immer";
+import { GameState } from "features/game/types/game";
+import { getKeys } from "features/game/types/decorations";
+import { getActiveCalenderEvent } from "features/game/types/calendar";
+import { BuildingName } from "features/game/types/buildings";
+
+export type TriggerTornadoAction = {
+  type: "tornado.triggered";
+};
+
+type Options = {
+  state: Readonly<GameState>;
+  action: TriggerTornadoAction;
+  createdAt?: number;
+};
+
+export function triggerTornado({
+  state,
+  action,
+  createdAt = Date.now(),
+}: Options) {
+  return produce(state, (game) => {
+    const tornado = game.calendar.dates.find(
+      (event) => event.name === "tornado",
+    );
+
+    if (!tornado) {
+      throw new Error("There is no tornado");
+    }
+
+    if (
+      createdAt - new Date(tornado.date).getTime() >
+      1000 * 60 * 60 * 24 * 3
+    ) {
+      throw new Error("Tornado is too old");
+    }
+
+    game.calendar.tornado = {
+      triggeredAt: createdAt,
+    };
+
+    const hasPinwheel = !!game.collectibles["Tornado Pinwheel"];
+
+    if (hasPinwheel) {
+      delete game.collectibles["Tornado Pinwheel"];
+      delete game.inventory["Tornado Pinwheel"];
```

Comment:
Are players not allowed to collect (inventory) multiple Tornado Pinwheel (from crafting or chests)?  Seems a bit limited to destroy all of them (even if not placed).

Reply:
Thanks @brian-man "Windmill" was a typo.

There is a limit of 1 Tornado Pinwheel per player. We don't want players to stock up on these 👍